strings/base_error.h

```diff
@@ -313,8 +313,8 @@ WINRT_EXPORT namespace winrt
             }
 
             com_ptr<impl::IErrorInfo> info;
-            WINRT_VERIFY_(0, WINRT_IMPL_GetErrorInfo(0, info.put_void()));
-            WINRT_VERIFY(info.try_as(m_info));
+            WINRT_IMPL_GetErrorInfo(0, info.put_void());
```

Comment:
`(void)` so it is clear to static analyzers and readers that the return code is intentionally ignored?

Reply:
Be nice if we could run said static analyzers as part of CI validation. I've never had much success getting Visual Studio's built in static analysis to flag anything in C++/WinRT.

Reply:
Some internal repos are set up to run Code Analysis during release builds (but not debug builds, because they add a lot of wallclock time).  The warnings/errors show up the same as normal compiler warnings/errors so it is hard to tell whether they are from static analysis or just the normal compiler.  Perhaps the same thing is happening here?

Reply:
I don't know. I just manually built this code with code analysis and I get nothing:

![image](https://github.com/microsoft/cppwinrt/assets/9845234/c7bc328b-4c6f-496d-a408-f0a9619908ab)

Reply:
I manually ran an analysis build (using the test_cpp20 project as the thing that pulls in base.h) and it only flagged a single warning using the default "Microsoft Native Recommended Rules".

V:\T\cppwinrt\_build\x64\Debug\winrt\base.h(3373): warning C26800: Use of a moved from object: ''(*<args_1>)'' (lifetime.1).

```cpp
#if __cpp_lib_format >= 202207L
    template <typename... Args>
    inline hstring base_format(Args&&... args)
    {
        auto const size = std::formatted_size(std::forward<Args>(args)...);
        WINRT_ASSERT(size < UINT_MAX);
        auto const size32 = static_cast<uint32_t>(size);

        hstring_builder builder(size32);
        WINRT_VERIFY_(size32, std::format_to_n(builder.data(), size32, std::forward<Args>(args)...).size);
        return builder.to_hstring();
    }
#endif
```

(WINRT_VERIFY_ is line 3373)

Reply:
I enabled "Run Code Analysis on Build" for the test_cpp20 project and rebuilt and it produced the same output.  It seems to work ok.

(There are also a handful of warnings in catch2 that I didn't copy off and we should probably exclude from consideration)

Reply:
To be fair I was a bit iffy about double forwarding stuff when I originally wrote this, but formatting *should* leave stuff untouched. I just wanted to avoid needing to copy stuff to format the string. Might change it to being `Args const&...` instead. Asking the STL devs for their opinion.